src/Algolia.Search/Transport/HttpTransport.cs

```diff
@@ -134,7 +137,7 @@ public HttpTransport(AlgoliaConfig config, IHttpRequester httpClient)
                 }
             }
 
-            throw new AlgoliaUnreachableHostException("Unreachable hosts");
+            throw new AlgoliaUnreachableHostException("RetryStrategy failed to connect to Algolia. Reason: " + errorMessage);
```

Comment:
IMO maybe the message is too specific. Maybe it's not necessary to talk about the `RetryStrategy` per se. Maybe something like. "Hosts are unreachable. Reason: + errorMessage" would be better. WDYT @TomKlotzPro @Ant-hem ?

Reply:
> maybe the message is too specific. 

From all ticket I've handle that's what our customer want, to provide as much detail as we could. 
Writing "Host are unreachable" on AlgoliaUnreachableHostException is the same thing. 

Explaining that our retry strategy failed due to a specific reason that cause an AlgoliaUnreachableHostException to be thrown is better suited. 

The initial was the the message which was unclear, now it's specific and details.

Reply:
TBH I have no strong opinion on the beginning of the error message, both are fine to me. What's really important is that we forward why the SDK failed to connect to our servers, this is useful for our users and for us while doing support.

@chloelbn and @TomKlotzPro we can discuss a better error message in the spec repository :) For the moment let's just merge this PR, we can still change the message afterwards as we don't commit on them.